src/Examine/LuceneEngine/WriterTracker.cs

```diff
@@ -41,9 +44,71 @@ public IndexWriter GetWriter(Directory dir, bool throwIfEmpty)
 
         public IndexWriter GetWriter(Directory dir, Func<Directory, IndexWriter> factory)
         {
-            var resolved = _writers.GetOrAdd(dir.GetLockId(), s => factory(dir));
-            return resolved;
+            lock (_locker)
```

Comment:
_writers is a ConcurrentDictionary, why do we need another lock? Else if we do because of the logic in MapAsOld, we can probably just make it a Dictionary and manage the locking ourselves?

Reply:
@Shazwazza I had some weird issue with just ConcurrentDictionary, so I added that, but will play around with just dictionary :)